middlewares/cachePolicy.js

```diff
@@ -3,18 +3,20 @@ const redis = require('redis');
 const REDISPORT = process.env.REDISPORT || 6379;
 const client = redis.createClient(REDISPORT);
 
-function cache(req,res,next){
-	const {code} = req.params;
+function cache(req, res, next) {
+	const { code } = req.params;
 
-	client.get(code,(err,data)=>{
-		if(err) throw err;
+	client.get(code, (err, data) => {
+		if (err) throw err;
 
-		 if(data!==null){
-			 console.log('Cache redirect',data);
-			  res.redirect(data); 
-		 }else{
-			 next();
-		 }
+		if (data !== null) {
+			console.log('Cache redirect', data);
+			req.found = true;
+			next()
+			// res.redirect(data);
+		} else {
+			next();
+		}
 	})
```

Comment:
what is this used for @BalgopalPatro ?

Reply:
I am checking if the pair key : long Url exist on redis or not if exist I am passing `found = true` in request to the next middleware. So that there we can check for it is found in cache so it will increase the counter and redirect to that url.
And if `found == false` then it will do as previous.